src/Control/Distributed/Process/Tests/CH.hs

```diff
@@ -1229,17 +1229,19 @@ testExitLocal TestTransport{..} = do
   localNode <- newLocalNode testTransport initRemoteTable
   supervisedDone <- newEmptyMVar
   supervisorDone <- newEmptyMVar
+  l <- newEmptyMVar
 
   pid <- forkProcess localNode $ do
-    (liftIO $ threadDelay 100000)
-      `catchExit` \_from reason -> do
+    (liftIO (putMVar l ()) >> expect) `catchExit` \_from reason -> do
         -- TODO: should verify that 'from' has the right value
         True <- return $ reason == "TestExit"
         liftIO $ putMVar supervisedDone ()
 
   runProcess localNode $ do
+    liftIO $ takeMVar l
     ref <- monitor pid
     exit pid "TestExit"
+    send pid ()
```

Comment:
Send is needed in order to shutdown process in order to not leave leftover processes.